pages/api/create-poap-events.js

```diff
@@ -0,0 +1,48 @@
+import { format, parse } from 'date-fns'
+import withMethods from '../../util/api/withMethods'
+import withAuth from '../../util/api/withAuth'
+import withValidParams from '../../util/api/withValidParams'
+import { createCommunityPoapEvent } from '../../util/api/fauna'
+import { getPoapEvent } from '../../util/api/poap'
+
+function _toFaunaDate(dateStr) {
+  return format(parse(dateStr, 'dd-MMM-yyyy', new Date()), 'yyyy-MM-dd')
+}
+
+const handler = async (req, res) => {
+  const { communityId, eventIds } = req.body
+
+  // Get POAP event data
+  await Promise.all(
+    eventIds.map(async eventId => {
+      const event = await getPoapEvent(eventId)
+      await createCommunityPoapEvent({
```

Comment:
This could be made more efficient by passing the array of event IDs to Fauna and implementing a custom UDF but I think this is fine for now.